gateway/gateway-requests/src/registration/handshake/gateway.rs

```diff
@@ -49,11 +50,28 @@ impl<'a> GatewayHandshake<'a> {
                 }
 
                 // init: <- pub_key || g^x
-                let (remote_identity, remote_ephemeral_key) = check_processing_error(
+                let init_message = check_processing_error(
```

Comment:
Please update this comment because it seems you're also sending the coconut credential alongside